android/build.gradle

```diff
@@ -139,6 +163,15 @@ android {
 
   buildFeatures {
     dataBinding = true
+    compose true
+  }
+
+  kotlinOptions {
+    jvmTarget = JavaVersion.VERSION_11
```

Comment:
```suggestion
    jvmTarget = JavaVersion.VERSION_17
```

Reply:
@Jean-BaptisteC did you test if this change will work properly on older Android 5 devices?

Reply:
For this to work we need to change this as well:
```
  compileOptions {
    sourceCompatibility JavaVersion.VERSION_17
    targetCompatibility JavaVersion.VERSION_17
  }
  ```
It compiles ok but I don't know the impact this will have.

Reply:
Tested on Lenovo Yoga Tab 2 Android 5.1, works but render is so long

Reply:
is it slower since the java 17 change or since adding compose? Compose should not impact this screen as it is only added on the about screen

Reply:
I guess Java 17, splash screen and load map is too long.

Reply:
If you tried the pr without changing it then it was not using java 17. I tried on an emulator running android 5 and could not find any difference. I don't have old device running this version so I'll need your help on this.

If its app loading time that's increased that's possible and that's exactly the type of stuff we want to test with this PR.

Can you please try and measure the startup difference between current release and this PR?

I'll try to measure that on the devices I have at hand.

@biodranik are releases and PR run artifacts optimized in the same way?

Reply:
Launch time (Load splashscreen + load map)
Latest release - 12,52 seconds
Debug compose with jvmTarget=17 - 33,12 seconds

Reply:
@arnaudvergnet fdroid beta should be equal to fdroid in the store. @Jean-BaptisteC comparing debug versions is not informative. Can you please compare fdroid beta from this branch with fdroid production start time?

Reply:
@Jean-BaptisteC why do you force jvmTarget=17? It works fine when using 11 as it is right now in the PR

Reply:
I tried on a Fairphone 3 and an emulator. Using master and this pr, with java 11 and 17, and I can't find any difference. It always takes around 2 to 5 seconds depending on the device.

Reply:
```
writeFdroidBetaSigningConfigVersions
ERROR: R8: java.lang.OutOfMemoryError: Java heap space
```

Reply:
Because of the libraries added we need to increase the heap space to something like 2GB (was at 1GB). It is optimized by R8 anyway but it needs more memory for compiling. I updated the PR with this change.

Reply:
Latest release FDroid - 11,78 seconds
Beta Fdroid - 9,50 seconds

Reply:
So its actually faster now in this PR? Don't know how its possible but that's nice :)

Reply:
It may be possible because of the system cache. A proper test is to reboot the device before each app run.

Reply:
Another reason is that there are already some commits after the public fdroid release was built. So the correct way to compare is to use this pr and then undo its commits and rebuild it again.

Reply:
In my tests I compared with the master version this pr is based on and could not find much difference.

Reply:
I have retry, load time between master and compose branch is similar ~9 seconds

Reply:
Please push Java 17 change separately. It is needed in any case.

Reply:
> compileOptions {
>     sourceCompatibility JavaVersion.VERSION_17
>     targetCompatibility JavaVersion.VERSION_17
>   }

Is this change compatible with Android 5 devices?

Reply:
Yes, i have try with Lenovo Tablet with Android 5.0